sql/log_event.cc

```diff
@@ -11261,12 +11261,17 @@ static enum_tbl_map_status check_table_map(Relay_log_info const *rli,
       rli->info_thd->system_thread == SYSTEM_THREAD_SLAVE_WORKER &&
       !thd_is_wsrep_applier && res == FILTERED_OUT) {
     Slave_worker *sw =
-        static_cast<Slave_worker *>(const_cast<Relay_log_info *>(rli));
-    Wsrep_async_monitor *wsrep_async_monitor{sw->get_wsrep_async_monitor()};
-    if (wsrep_async_monitor) {
-      auto seqno = sw->sequence_number();
-      assert(seqno > 0);
-      wsrep_async_monitor->skip(seqno);
+        dynamic_cast<Slave_worker *>(const_cast<Relay_log_info *>(rli));
+    // It should never happen. If this is SYSTEM_THREAD_SLAVE_WORKER, but it
+    // is not wsrep applier, it has to be Slave_worker.
+    assert(sw != nullptr);
+    if (sw) {
+      Wsrep_async_monitor *wsrep_async_monitor{sw->get_wsrep_async_monitor()};
+      if (wsrep_async_monitor) {
+        auto seqno = sw->sequence_number();
+        assert(seqno > 0);
+        wsrep_async_monitor->skip(seqno);
+      }
     }
```

Comment:
on release builds, what to do the else part? can we continue safely if such cast gives nullptr ? or better to abort with unireg_abort?

Reply:
it is like "this should never happen" path. `rli->info_thd->system_thread == SYSTEM_THREAD_SLAVE_WORKER &&
      !thd_is_wsrep_applier` enforces it to never be nullptr after dynamic cast. We could use static_cast here. This is just playing safe.
But if anything still goes wrong, I think it should go on, not abort. In such a case it would work as without the fix, so still low chances of hitting the deadlock.

Reply:
may be ask the user to turn off the feature and restart? :thinking: async monitor can be disabled?